src/main/java/uk/gov/pay/products/service/LinksBuilder.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.pay.products.service;
+
+import uk.gov.pay.products.model.Link;
+import uk.gov.pay.products.model.Product;
+
+import java.net.URI;
+
+import static javax.ws.rs.HttpMethod.GET;
+import static javax.ws.rs.HttpMethod.POST;
+import static javax.ws.rs.core.UriBuilder.fromUri;
+import static uk.gov.pay.products.resources.ProductResource.PRODUCTS_RESOURCE;
+
+public class LinksBuilder {
```

Comment:
Calling this builder with a method decorate is not intuitive, maybe rename this to LinksDecorator? or something like that?